### PR TITLE
 Improve logging for kubernetes API communication issues

### LINKED ISSUE #371: Kubernetes service discovery: Log k8s API entity if a non-200 response code
Atm it logs the stack trace which is quite useless as it just says it can't unmarshall the body.

If there is a non-200 we shouldn't try and unmarshall the body and just log it at ERROR. 

Currently to get a useful log DEBUG is required.

### PR DESCRIPTION
**Pull request by chbatey**
* Don't unmarshal the response for non-200 status codes
* Specific log message for 403
* Log body at error for non-200 so users don't need to turn on DEBUG

Fixes #371

### CODE REVIEW

File: discovery-kubernetes-api/src/main/scala/akka/discovery/kubernetes/KubernetesApiSimpleServiceDiscovery.scala [hunk 1]
```diff
@@ -99,16 +103,28 @@ class KubernetesApiSimpleServiceDiscovery(system: ActorSystem) extends SimpleSer
...
 
-        unmarshalled.failed.foreach { t =>
-          log.error(t, "Failed to unmarshal Kubernetes API response status [{}]; check RBAC settings",
-            response.status.value)
+        response.status match {
+          case StatusCodes.OK =>
+            log.debug("Kubernetes API entity: [{}]", entity.data.utf8String)
+            val unmarshalled = Unmarshal(entity).to[PodList]
+            unmarshalled.failed.foreach { t =>
+              log.error(t, "Failed to unmarshal Kubernetes API response.  Status code: [{}]; Response body: [{}]",
```
Comment (raboof) [inline discovery-kubernetes-api/src/main/scala/akka/discovery/kubernetes/KubernetesApiSimpleServiceDiscovery.scala]: :+1:

Comment (raboof) [reply discovery-kubernetes-api/src/main/scala/akka/discovery/kubernetes/KubernetesApiSimpleServiceDiscovery.scala]: Maybe 'error' is a bit strong? At least 'warning' though indeed

File: discovery-kubernetes-api/src/main/scala/akka/discovery/kubernetes/KubernetesApiSimpleServiceDiscovery.scala [hunk 2]
```diff
@@ -99,16 +103,28 @@ class KubernetesApiSimpleServiceDiscovery(system: ActorSystem) extends SimpleSer
...
+                response.status.value, entity)
```
Comment (raboof) [inline discovery-kubernetes-api/src/main/scala/akka/discovery/kubernetes/KubernetesApiSimpleServiceDiscovery.scala]: does `entity.toString` contain the whole response or is it truncated?

File: discovery-kubernetes-api/src/main/scala/akka/discovery/kubernetes/KubernetesApiSimpleServiceDiscovery.scala (hunk 1 above)
Comment (chbatey) [reply discovery-kubernetes-api/src/main/scala/akka/discovery/kubernetes/KubernetesApiSimpleServiceDiscovery.scala]: I'll switch to warning. The only case now is a version mis-match so nothing is likely to work.

File: discovery-kubernetes-api/src/main/scala/akka/discovery/kubernetes/KubernetesApiSimpleServiceDiscovery.scala (hunk 2 above)
Comment (chbatey) [reply discovery-kubernetes-api/src/main/scala/akka/discovery/kubernetes/KubernetesApiSimpleServiceDiscovery.scala]: It'll truncate. Will switch to Unmarshal as a String.

**Pull request merged**